lib/internal/fs/sync_write_stream.js

```diff
@@ -19,7 +19,21 @@ Object.setPrototypeOf(SyncWriteStream.prototype, Writable.prototype);
 Object.setPrototypeOf(SyncWriteStream, Writable);
 
 SyncWriteStream.prototype._write = function(chunk, encoding, cb) {
-  writeSync(this.fd, chunk, 0, chunk.length);
+  while (true) {
+    try {
+      const n = writeSync(this.fd, chunk, 0, chunk.length);
+      if (n !== chunk.length) {
+        chunk = chunk.slice(0, n);
```

Comment:
Are the last bytes written first? AFAIK this should be `chunk = chunk.slice(n)`? Maybe just verify that in the test as well?

Reply:
@mcollina PTAL. This seems wrong to me.